samples/nrf9160/modem_trace_flash/src/trace_flash.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/storage/stream_flash.h>
+#include <zephyr/storage/flash_map.h>
+#include <zephyr/logging/log.h>
+#include <modem/trace_backend.h>
+
+#include "trace_flash.h"
+
+LOG_MODULE_REGISTER(modem_trace_flash_backend, CONFIG_MODEM_TRACE_FLASH_BACKEND_LOG_LEVEL);
+
+#define EXT_FLASH_DEVICE DEVICE_DT_GET(DT_ALIAS(ext_flash))
+#define TRACE_OFFSET FLASH_AREA_OFFSET(modem_trace)
+#define TRACE_SIZE FLASH_AREA_SIZE(modem_trace)
+#define BUF_LEN 1024
+
+static const struct flash_area *modem_trace_area;
+static const struct device *flash_dev;
+static struct stream_flash_ctx stream;
+static uint8_t stream_buf[BUF_LEN];
+
+// Persist trace size in .noinit RAM to survive warm boot.
+static __noinit uint32_t magic;
+static __noinit uint32_t traces_size;
```

Comment:
```suggestion
static uint32_t traces_size;
```

We dont need the `magic` variable now as we dont need to know if this is the first start of the app (this was the original design but we deviated from that idea). We also do not need to persist trace_size across reboots